flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/UserDefinedFunctionUtils.scala

```diff
@@ -358,4 +366,116 @@ object UserDefinedFunctionUtils {
     InstantiationUtil
       .deserializeObject[UserDefinedFunction](byteData, Thread.currentThread.getContextClassLoader)
   }
+
+  /**
+    * Build a TableFunctionCall, a name and a sequence of params will determine a unique
+    * [[TableFunctionCall]]
+    *
+    * @param name function name
+    * @param implicitResultType If no result type returned, it will use this type.
+    * @param params The input expressions
+    * @return A unique [[TableFunctionCall]]
+    */
+  private[table] def buildTableFunctionCall(
+      name: String,
+      tableFunction: TableFunction[_],
+      implicitResultType: TypeInformation[_],
+      params: Expression*): TableFunctionCall = {
+    val arguments = expressionsToArguments(params: _*)
+    val userDefinedResultType = tableFunction.getResultType(arguments)
+    val resultType = {
+      if (userDefinedResultType != null) userDefinedResultType
+      else implicitResultType
+    }
+    TableFunctionCall(name, tableFunction, params, resultType)
+  }
+
+  /**
+    * Transform the expressions or rex nodes to Objects
+    * Only literal expressions will be passed, nulls for non-literal ones
+    *
+    * @param params actual parameters of the function
+    * @return A java List of the Objects
+    */
+  private[table] def expressionsToArguments(params: Expression*): java.util.List[AnyRef] = {
+    params.map {
+      case exp: Literal =>
+        exp.value.asInstanceOf[AnyRef]
+      case _ =>
+        null
+    }
+  }
+
+  /**
+    * Transform the rex nodes to Objects
+    * Only literal expressions will be passed, nulls for non-literal ones
+    *
+    * @param rexNodes actual parameters of the function
+    * @return A java List of the Objects
+    */
+  private[table] def rexNodesToArguments(
+      rexNodes: java.util.List[RexNode]): java.util.List[AnyRef] = {
+    rexNodes.map {
+      case rexNode: RexLiteral =>
```

Comment:
Can we add an unit test to ensure all supported literal are successfully transformed? And other non supported type will translate to null.